pkgs/development/python-modules/ihatemoney/default.nix

```
@@ -103,47 +78,41 @@ buildPythonPackage rec {
     pytz
     requests
     six
-    (
-      (
-        sqlalchemy-continuum.override {
-          sqlalchemy = pinned_sqlalchemy;
-          sqlalchemy-utils = sqlalchemy-utils.override {
-            sqlalchemy = pinned_sqlalchemy;
-          };
-          sqlalchemy-i18n = sqlalchemy-i18n.override {
-            sqlalchemy = pinned_sqlalchemy;
-            sqlalchemy-utils = sqlalchemy-utils.override {
-              sqlalchemy = pinned_sqlalchemy;
-            };
-          };
-          flask_sqlalchemy = flask_sqlalchemy.override {
-            sqlalchemy = pinned_sqlalchemy;
-          };
-        }
-      ).overridePythonAttrs (
-        old: {
-          doCheck = false;
-        }
-      )
-    )
+    sqlalchemy-continuum
     werkzeug
     wtforms
     psycopg2
     debts
   ];
 
-  # upstream performed the update without needing to patch the code
-  # the original patch does not apply, sadly
-  # https://github.com/spiral-project/ihatemoney/pull/912
+  patches = [
+    # fix build with wtforms 3. remove with next release
+    (fetchpatch {
+      url = "https://github.com/spiral-project/ihatemoney/commit/40ce32d9fa58a60d26a4d0df547b8deb709c330d.patch";
+      sha256 = "sha256-2ewOu21qhq/AOZaE9qrF5J6HH0h6ohFgjDb+BYjJnuQ=";
+      excludes = [ "setup.cfg" ];
+    })
+  ];
+
   postPatch = ''
+    # upstream performed the update without needing to patch the code
+    # the original patch does not apply, sadly
+    # https://github.com/spiral-project/ihatemoney/pull/912
     substituteInPlace setup.cfg --replace "Flask-WTF>=0.14.3,<1" "Flask-WTF>=0.14.3,<2"
+    # https://github.com/spiral-project/ihatemoney/pull/986
+    substituteInPlace setup.cfg --replace "SQLAlchemy>=1.3.0,<1.4" "SQLAlchemy>=1.3.0,<1.5"
+    # enabled by the patch above, remove on next release
+    substituteInPlace setup.cfg --replace "WTForms>=2.3.1,<2.4" "WTForms"
+
   '';
```

```suggestion
    substituteInPlace setup.cfg \
      --replace "Flask-WTF>=0.14.3,<1" "Flask-WTF>=0.14.3,<2" \
      --replace "SQLAlchemy>=1.3.0,<1.4" "SQLAlchemy>=1.3.0,<1.5" \
      --replace "WTForms>=2.3.1,<2.4" "WTForms"
  '';
```
super common to do in python and doing it with patches is discouraged because they easily break.